quesma/quesma/ui/login.go

```diff
@@ -0,0 +1,60 @@
+// Copyright Quesma, licensed under the Elastic License 2.0.
+// SPDX-License-Identifier: Elastic-2.0
+
+package ui
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"net/http"
+	"quesma/elasticsearch"
+)
+
+func (qmc *QuesmaManagementConsole) generateLoginForm() []byte {
+	buffer := newBufferWithHead()
+	buffer.Html(`<html>`)
+	buffer.Html(`<body>`)
+	buffer.Html(`<div class="login-screen">`)
+	buffer.Html(`<div class="login-form">`)
+	buffer.Html(`<h2>Login</h2>`)
+	buffer.Html(`<p style="color: #ccc;">Log in to Quesma admin console using your Elasticsearch credentials</p>`)
+	buffer.Html(`<form action="/login-with-elasticsearch" method="post">`)
+	buffer.Html(`<label for="username">Username:</label>`)
+	buffer.Html(`<input type="text" id="username" name="username" placeholder="Enter your Elasticsearch username">`)
+	buffer.Html(`<label for="password">Password:</label>`)
+	buffer.Html(`<input type="password" id="password" name="password" placeholder="Enter your Elasticsearch password">`)
+	buffer.Html(`<input type="submit" value="Login">`)
+	buffer.Html(`</form>`)
+	buffer.Html(`</div>`)
+	buffer.Html(`</div>`)
+	buffer.Html(`</body>`)
+	buffer.Html(`</html>`)
+	return buffer.Bytes()
+}
+
+func (qmc *QuesmaManagementConsole) HandleElasticsearchLogin(writer http.ResponseWriter, req *http.Request) {
+	if req.Method == http.MethodGet {
+		writer.Header().Set("Content-Type", "text/html")
+		writer.Write(qmc.generateLoginForm())
+	} else if req.Method == http.MethodPost {
+		username := req.FormValue("username")
+		password := req.FormValue("password")
+		if qmc.isValidElasticsearchUser(username, password) {
+			session, _ := store.Get(req, quesmaSessionName)
```

Comment:
Add an error log here. It's tricky to debug login issues.

Reply:
https://github.com/QuesmaOrg/quesma/pull/836/commits/2dee16ae8063e7e2d755e8925b49bd4752a6e2ae